global-script.lua

```diff
@@ -3088,4 +3148,87 @@ function tCompare(t1,t2)
     return table.concat(newTab,"|")
     end
     if cc2(t1) == cc2(t2) then return true else return false end
-end
\ No newline at end of file
+end
+
+
+function swapPlayerAreas(a, b)
+    local function tableSwap(table)
+        local temp = table[a]
+        table[a] = table[b]
+        table[b] = temp
+    end
+    local function tintSwap(table)
+        local oa = table[a]
+        local ob = table[b]
+        if type(oa) == "string" then
+            oa = getObjectFromGUID(oa)
+            ob = getObjectFromGUID(ob)
+        end
+        local ta = oa.getColorTint()
+        local tb = ob.getColorTint()
+        oa.setColorTint(tb)
+        ob.setColorTint(ta)
+        tableSwap(table)
+    end
+    local function positionSwap(table)
+        local oa = table[a]
+        local ob = table[b]
+        if type(oa) == "string" then
+            oa = getObjectFromGUID(oa)
+            ob = getObjectFromGUID(ob)
+        end
+        local ta = oa.getPosition()
+        local tb = ob.getPosition()
+        oa.setPosition(tb)
+        ob.setPosition(ta)
+    end
+
+    print("Player " .. a .. " is trading places with player " .. b .. ".")
+    for i = 1,2 do
+        local ta = Player[a].getHandTransform(i)
+        local tb = Player[b].getHandTransform(i)
+        Player[a].setHandTransform(tb, i)
+        Player[b].setHandTransform(ta, i)
+    end
+    tintSwap(playerTables)
+    tableSwap(playerBlocks)
+    positionSwap(defendBags)
+    positionSwap(readyTokens)
+    tableSwap(elementScanZones)
+    if playerBlocks[a] then playerBlocks[a].setVar("playerColor", a) end
+    if playerBlocks[b] then playerBlocks[b].setVar("playerColor", b) end
+end
```

Comment:
don't you need to move all of the elementBags for the player table as well? otherwise the the element counting would go to the wrong player right?

Reply:
Nope!  The elementBags count the tokens in their zone.  The zones and player blocks aren't actually moved, we just change which zone is considered to belong to which player -- hence these assignments needing to be saved.  Since we don't move the zones, we also don't need to move the element bags.

As it turns out, nothing currently cares whose zone is whose: The Element Helper just sums all 6 colors worth, and the individual player blocks just count how many are in their associated zone.  Thus, the `tableSwap(elementScanZones)` isn't really required unless something down the line wants to be able to know "How much fire does Red have" (as opposed to "How much fire does the first player slot have" or "How much fire does everyone have")

An earlier version of my code DID move the elementScanZones, which was nice for save/load but did the counting in the wrong place.

Reply:
Correction: Gain A Spirit cares whose `elementScanZone` it is.